compile error with AOCC flang with profiling enabled

With the new v2.0.2 release, the library can be built with AOCC. 

However, when compiling 2decomp-fft using AOCC flang with profiling enabled, the following errors are reported:

```
F90-S-0038-Symbol, decomp_profiler_none, has not been explicitly declared (/local/ningli12/spack-stage/spack-stage-2decomp-fft-2.0.2-ivqrs6siv4fmcvviukflvofl23wnitbq/spack-src/src/profiler_caliper.f90)
  0 inform,   0 warnings,   1 severes, 0 fatal for decomp_profiler_fin_noarg
F90-S-0038-Symbol, decomp_profiler_caliper, has not been explicitly declared (/local/ningli12/spack-stage/spack-stage-2decomp-fft-2.0.2-ivqrs6siv4fmcvviukflvofl23wnitbq/spack-src/src/profiler_caliper.f90)
  0 inform,   0 warnings,   1 severes, 0 fatal for decomp_profiler_prep_bool
make[2]: *** [src/CMakeFiles/decomp2d.dir/build.make:182: src/CMakeFiles/decomp2d.dir/profiler_caliper.f90.o] Error 1
```

This looks like a compiler bug. It is similar to but not the same as the issue in #245 

It looks like the compiler is unable to access private enum data in a module from submodules. Before I report this to the compiler developers, can you please check the following:

- Does the code conform to the Fortran standard?
- Is there a simple fix similar to the fix for #245?

--- a/src/decomp_2d_constants.f90
+++ b/src/decomp_2d_constants.f90
@@ -74,10 +74,8 @@ module decomp_2d_constants
    !    0 => no profiling, default
    !    1 => Caliper (https://github.com/LLNL/Caliper)
    !
-   enum, bind(c)
-      enumerator :: decomp_profiler_none = 0
-      enumerator :: decomp_profiler_caliper = 1
-   end enum
+   integer, parameter, public :: DECOMP_PROFILER_NONE = 0
+   integer, parameter, public :: DECOMP_PROFILER_CALIPER = 1
 
    !
    ! Supported FFT backends

